src/main/java/ch/njol/skript/expressions/base/ExprListIndex.java

```diff
@@ -0,0 +1,85 @@
+package ch.njol.skript.expressions.base;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.Variable;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.LiteralUtils;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+public class ExprListIndex extends SimpleExpression<Object> {
+
+	static {
+		Skript.registerExpression(ExprListIndex.class, Object.class, ExpressionType.COMBINED,
+			"[(first|2¦last)] [(1¦variable)] index of %object% in %objects%",
+			"[all] [(1¦variable)] ind(ex|ic)es of %object% in %objects%");
```

Comment:
We could add support for multiple objects in the second syntaxes index search

Reply:
I will leave you guys to it